### ISSUE TITLE
Install CSI plugin for storage

### ISSUE THREAD

**Issue by buggtb** [unity-sds/unity-cs#144]

**Comment by buggtb** [comment]
blocked by MCP

**Comment by pymonger** [comment]
I ran into this issue too so I installed the plugin using kubectl: https://github.com/unity-sds/unity-on-demand-cloudformation/blob/b17b793ad265e1d3e89f5fe0034bdfdc14849c91/templates/unity_deployer_instance-act.yaml#L249

**Comment by buggtb** [comment]
yeah the installing bit isn't the hard part. The refusal of being allowed to automatically configure an OIDC connection so the plugin can manage the volumes in the cluster is the bit that blocks it up.

**Comment by pymonger** [comment]
Exactly. When I tried to install the plugin through the EKS dashboard so
that EKS manages the CSI plugin, I ran into that OIDC issue. Instead of
going through that red tape with MCP, simply running  that kubectl command,
if I understand it correctly, installs that  plugin manually without
needing to configure OIDC connectors. I think it just  uses the
mcp-tenantOperator permissions that are "inherited" by the cluster when
created by a user with mcp-tenantOperator. In any case I was able to
configure EBS CSI on EKS clusters in a non-Unity MCP account as well as the
Unity dev account.

**Comment by buggtb** [comment]
Hrm, the OIDC stuff still causes us a whole bunch of security woes, but I'm curious to know if it actually used the plugin or not because I tried to spin up a 1.24 cluster which is where its required in MCP and there's no images for the VMs new enough and I couldn't attach any nodes. If you use 1.21 for example the old plugin(non OIDC) is still in the codebase and so the addon plugin isn't required

**Comment by buggtb** [comment]
Thanks Gerald, I used your ami's that worked well. I spoke with JPL cyber who said that the CNI plugin for EBS will provision volumes because our access policy is too broad and effectivley causes security issues in allowing access to resources from the nodes. So it does work, but i'm still going to continue to attempt to get MCP to actually fix this stuff.

**Comment by buggtb** [comment]
Okay well the plugin installs, the PV provisioning works but only because of lax security thats covered in another ticket so i'll close this.
